static/app/components/events/eventTagsAndScreenshot/tags.tsx

```diff
@@ -23,13 +25,33 @@ type Props = {
   projectSlug: Project['slug'];
 };
 
+function filterTags(tags: EventTag[], tagFilter: TagFilter): EventTag[] {
+  switch (tagFilter) {
+    case TagFilter.ALL:
+      return tags;
+    case TagFilter.CUSTOM:
+      return tags.filter(tag => !SentryDefaultTags.has(tag.key));
+    default:
+      return tags.filter(tag => TagFilterData[tagFilter].has(tag.key));
+  }
+}
+
 function Tags({event, projectSlug}: Props) {
   const [tagFilter, setTagFilter] = useState<TagFilter>(TagFilter.ALL);
-  const handleTagFilterChange = useCallback((value: TagFilter) => {
-    setTagFilter(value);
-  }, []);
+  const [tags, setTags] = useState<EventTag[]>(filterTags(event.tags ?? [], tagFilter));
```

Comment:
this could be a useMemo that changes on tagFilter + event.tags, i think an initial state that relies on another initial state can be confusing to figure out later